### ISSUE TITLE
[ENHANCEMENT] Font toolbar for individual drop caps and lyric boxes

### ISSUE THREAD

**Issue by cnkaram** [danielgarthur/neanes#67]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
I would like the ability to change the drop cap Font and color as well as the lyric font, text size, and color

**Describe alternatives you've considered**
I couldn't find any 

**Additional context**

**Comment by danielgarthur** [comment]
Adjusting the font for drop caps and lyrics can be done in the Page Setup dialog (File -> Page Setup) or Ctrl+Shift+P. This affects all the lyrics and drop caps for the entire document.

<img width="412" alt="image" src="https://user-images.githubusercontent.com/4132779/222803575-edb8db5d-b361-438d-82af-6a5ad84209a7.png">

If you want to change the font for a single lyric box or for a single drop cap, that is not currently supported, but could be added in a later release if that's what you are wanting. Are you wanting a toolbar on the bottom like this one for individual drop caps and lyric boxes?

<img width="574" alt="image" src="https://user-images.githubusercontent.com/4132779/222803859-5c9fa5b9-7177-4826-96d4-7cfaa1567867.png">

**Comment by cnkaram** [comment]
Thank you for the reply and the beautiful work. A toolbar to change the font of the drop cap and the text would make things easier, and I think would be a great addition to this program.
I have another concern. If, God forbids, the program is discontinued or you come up with another version that is not compatible with the current one, what happens to the my music projects prepared with this version. And since the output is a PDF, I won't have an editable document of the projects but just a PDF. This means I will have to type it again in another program. Is that a valid concern or am I over thinking? Besides that, it is a beautiful and and a user friendly program.
Thank you again,Chadi

Sent from Yahoo Mail on Android 
 
  On Fri, Mar 3, 2023 at 12:57 PM, ***@***.***> wrote:   


Adjusting the font for drop caps and lyrics can be done in the Page Setup dialog (File -> Page Setup) or Ctrl+Shift+P. This affects all the lyrics and drop caps for the entire document.



If you want to change the font for a single lyric box or for a single drop cap, that is not currently supported, but could be added in a later release if that's what you are wanting. Are you wanting a toolbar on the bottom like this one for individual drop caps and lyric boxes?

**Comment by danielgarthur** [comment]
To answer your concern about the versioning, when I make changes to the file format, I attempt to make the newer version of the program able to read old files. I have also considered turning off the auto-update feature, in case there is a bug or breaking change in a newer version. Right now, on Windows and Linux, if a newer version is available, the update is automatic. But I think that instead giving the user a choice to upgrade or keep the current version could make more sense.

As to your other concern about not having an editable file, I think there's a lot to cover there. The short answer is that I understand your concern and I would have that concern, too.

May I ask what your current process is for writing music? Do you use Microsoft Word? If so, which fonts do you use and how to you position the neumes precisely and how do you align the text? 

With some more information, I could think about whether there might be some kind of middle ground such as an export to Word. I have also been considering an export to HTML for some time, but it is still in the design phase. The idea is that Neanes files or parts of a file could be exported as HTML and embedded in any web page.

For some background, I created Neanes because at the time, the only option for writing Byzantine music on a computer in English that I knew of was to use Microsoft Word and some special font sets, together with macros and various hacks to get the positioning correct. These hacks may not always work in future version of Word. Positioning of lyrics was done through custom text box elements. And each font had a different set of keys that you had to memorize to type the correct characters, and the macros did not necessarily work for different fonts. You also had to constantly switch between different fonts to get all the characters. There is also the concern that Microsoft has made a shift in recent years to the cloud-based Office 365, and I'm not sure how well Byzantine fonts would work on a browser-only Microsoft Word app. So even the idea of using Microsoft Word with its flaws gave me the same concerns about the longevity of old files. 

I wanted something easy to use that functioned like Musescore (or Finale or Sibelius). I created Neanes and SBMuFL hoping to inspire others by demonstrating how something like this might be done. SBMuFL (inspired by SMuFL) is my other project, which attempts to standardize Byzantine neume fonts. It allows the use of a single font that contains all the characters, as well as some automatic positioning of the supporting characters such as the gorgon and fthores. The trade-off is that one can no longer use the keyboard to enter all the characters. Instead some kind of intermediate software such as Neanes would be necessary to access the characters in the font. 

If there comes a time when I am no longer able to work on these projects, I hope that someone else could take up this work, which is why all the code is open source. Ideally, other experienced developers who know Byzantine chant would get involved before that time.

**Comment by cnkaram** [comment]
Thank you for your reply and explanation.
I really like the Neanes software. For years, I have been using Microsoft Word with the Fonts and Template created by Fr.Ephraim from St.Anthony's monastery. 

You are right about what you wrote regarding the macros, positioning of neumes and the text under them. It is very time consuming. Sometimes, I have to manually position both the lyrics and the neumes depending on how big the word/syllable is. I typed and still typing all my music compositions using Microsoft Word. The issue I am facing now is, when I open a document typed using an older Microsoft Word version, the macros do not work, I have to copy the music into a new template in order for the Macros to work. I am not sure why this is happening. And now you made  me concerned about the cloud-based Office 365, but so far everything seems to be working fine.

I always wanted  to use a program like Finale. My friend uses it to adapt my compositions to Western notation. I am amazed at how user-friendly Finale is. I think the Neanes program has all the features I see in Finale and needed to write Byzantine music.What you said about the possibility to export the work to Word/HTML sounds good, but don't I need the fonts installed in order for Word to read the neumes?

Thank you again for addressing my concerns and for creating the Neanes software.
Chadi 
 

    On Monday, March 6, 2023 at 02:56:38 PM CST, danielgarthur ***@***.***> wrote:  
 
 


To answer your concern about the versioning, when I make changes to the file format, I attempt to make the newer version of the program able to read old files. I have also considered turning off the auto-update feature, in case there is a bug or breaking change in a newer version. Right now, on Windows and Linux, if a newer version is available, the update is automatic. But I think that instead giving the user a choice to upgrade or keep the current version could make more sense.

As to your other concern about not having an editable file, I think there's a lot to cover there. The short answer is that I understand your concern and I would have that concern, too.

May I ask what your current process is for writing music? Do you use Microsoft Word? If so, which fonts do you use and how to you position the neumes precisely and how do you align the text?

With some more information, I could think about whether there might be some kind of middle ground such as an export to Word. I have also been considering an export to HTML for some time, but it is still in the design phase. The idea is that Neanes files or parts of a file could be exported as HTML and embedded in any web page.

For some background, I created Neanes because at the time, the only option for writing Byzantine music on a computer in English that I knew of was to use Microsoft Word and some special font sets, together with macros and various hacks to get the positioning correct. These hacks may not always work in future version of Word. Positioning of lyrics was done through custom text box elements. And each font had a different set of keys that you had to memorize to type the correct characters, and the macros did not necessarily work for different fonts. You also had to constantly switch between different fonts to get all the characters. There is also the concern that Microsoft has made a shift in recent years to the cloud-based Office 365, and I'm not sure how well Byzantine fonts would work on a browser-only Microsoft Word app. So even the idea of using Microsoft Word with its flaws gave me the same concerns about the longevity of old files.

I wanted something easy to use that functioned like Musescore (or Finale or Sibelius). I created Neanes and SBMuFL hoping to inspire others by demonstrating how something like this might be done. SBMuFL (inspired by SMuFL) is my other project, which attempts to standardize Byzantine neume fonts. It allows the use of a single font that contains all the characters, as well as some automatic positioning of the supporting characters such as the gorgon and fthores. The trade-off is that one can no longer use the keyboard to enter all the characters. Instead some kind of intermediate software such as Neanes would be necessary to access the characters in the font.

If there comes a time when I am no longer able to work on these projects, I hope that someone else could take up this work, which is why all the code is open source. Ideally, other experienced developers who know Byzantine chant would get involved before that time.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
  

|  | Virus-free.www.avast.com |

**Comment by danielgarthur** [comment]
Yes, in order for Word to read the neumes, you would need the fonts installed. That means one of two things:

1. Neanes could export to a Word document that uses the old set of fonts with the proper adjustments, so at least they documents are editable as before. 
2. Neanes could export to a Word document that uses the new single font, but it would be really difficult to edit the document by hand.

I've spent some time this last week putting together a library of custom web components. It allows neumes to be put into any webpage in a straightforward way. The only downside is that it's tedious to type out the markup. But I plan to allow an export from Neanes to HTML to speed up the process. The HTML library is not dependent on Neanes, and HTML will probably be here for many years to come. So in a way the HTML library is more resilient to change than Neanes, I think.

One interesting bonus to using HTML is that all the features of modern web pages are available, such as responsiveness. For example, check out the samples here: [Sample 1 - Greek](https://danielgarthur.github.io/byzhtml/#/sample1) and [Sample 2 - English](https://danielgarthur.github.io/byzhtml/#/sample2). Try opening them on a laptop/desktop, then on a tablet, then on a phone. Try resizing the window on the desktop. Try turning the phone sideways, putting it into landscape mode. The neumes instantly re-arrange and yet stay justified for easy reading.

This is much easier than trying to read a static PDF on a phone and needing to scroll left and right with your finger and zoom in really far to see the neumes. However, I do still plan to include an option in Neanes to export exactly as it was displayed in Neanes, without any fancy auto-adjusting. This would be an editable artifact in HTML format that looks exactly like the document in Neanes.

One last thing, the latest release now allows drop cap styles to be edited individually.  Lyric styles should be coming soon.

**Comment by danielgarthur** [comment]
Added to release `0.4.3`.

**Comment by basil** [comment]
> I wanted something easy to use that functioned like Musescore (or Finale or Sibelius).

Finale is reaching end of life after 35 years: https://www.finalemusic.com/blog/end-of-finale-new-journey-dorico-letter-from-president/

> If there comes a time when I am no longer able to work on these projects, I hope that someone else could take up this work, which is why all the code is open source.

The only true way to ensure longevity of data is to use open data formats and open-source software. There is a petition to release Finale to the open source community: https://www.change.org/p/release-finale-music-notation-software-to-the-open-source-community

Western music notation software has been plagued with abrupt endings like this, where a proprietary program stops receiving updates and users are forever locked out of their scores (or forced to run older versions in emulators). Previous examples are Leland Smith's SCORE and MOTU's Mosaic, both excellent pieces of software whose source code remains forever lost to the sands of time.

There may come a time when Melodos becomes unmaintainable as proprietary software, for any number of reasons (economic, personal, technical, etc). I, for one, am investing my time and resources into open-source Byzantine notation software.

**Comment by cnkaram** [comment]
Hello,
Can you please explain the purpose of this email? You probably have a record of the emails I sent a while ago. I asked about the possibility to add a feature that allows to export the score in a Word document in addition to PDF. I am aware that this requires you to provide the fonts, but I think that this function is important in case anything happens to the Neanes program and, for some reason, gets discontinued. If I have the Word document, I would be able to edit and not re-type as it would happen if only the PDF is available. This is my only hesitation for not using the Neanes software yet.

Chadi 
 

    On Tuesday, August 27, 2024 at 07:31:08 PM CDT, Basil Crow ***@***.***> wrote:  
 
 



I wanted something easy to use that functioned like Musescore (or Finale or Sibelius).


Finale is reaching end of life after 35 years: https://www.finalemusic.com/blog/end-of-finale-new-journey-dorico-letter-from-president/


If there comes a time when I am no longer able to work on these projects, I hope that someone else could take up this work, which is why all the code is open source.


The only true way to ensure longevity of data is to use open data formats and open-source software. There is a petition to release Finale to the open source community: https://www.change.org/p/release-finale-music-notation-software-to-the-open-source-community

Western music notation software has been plagued with abrupt endings like this, where a proprietary program stops receiving updates and users are forever locked out of their scores (or forced to run older versions in emulators). Previous examples are Leland Smith's SCORE and MOTU's Mosaic, both excellent pieces of software whose source code remains forever lost to the sands of time.

There may come a time when Melodos becomes unmaintainable as proprietary software, for any number of reasons (economic, personal, technical, etc). I, for one, am investing my time and resources into open-source Byzantine notation software.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
  

|  | Virus-free.www.avast.com |

**Comment by basil** [comment]
If anything, Finale's transition to an end-of-life product offers an opportunity to reflect on the sustainability of music notation software more broadly. There are enough Orthodox Christians in existence for there to always be demand for a Byzantine notation scorewriter, so I do not anticipate the need to return to word processing software. But just as in the Finale to Dorico migration, interoperability is a concern—to say nothing of the economic sustainability of proprietary scorewriters in a niche market.

On the positive side, Neanes is open-source software whose data format is simple JSON—an enormous advantage. On the negative side, the Neanes data format carries some legacy cruft and reflects internal implementation details, making it unsuitable as a general-purpose data format for other Byzantine notation scorewriters.

Here there is a lesson to be learned from the Finale to Dorico transition regarding the importance of open, vendor-neutral data formats—just as MusicXML created a path to export data from Finale and import data into Dorico, so too could a vendor-neutral data format for Byzantine notation scorewriters support e.g. export from Melodos and e.g. import into Neanes.

**Comment by cnkaram** [comment]
Ok. Thanks for the explanation. Is it hard to add the option to export the file as a Word document and provide the fonts? Is it not doable? 

Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, Sep 7, 2024 at 2:56 PM, Basil ***@***.***> wrote:   


If anything, Finale's transition to an end-of-life product offers an opportunity to reflect on the sustainability of music notation software more broadly. There are enough Orthodox Christians in existence for there to always be demand for a Byzantine notation scorewriter, so I do not anticipate the need to return to word processing software. But just as in the Finale to Dorico migration, interoperability is a concern—to say nothing of the economic sustainability of proprietary scorewriters in a niche market.

On the positive side, Neanes is open-source software whose data format is simple JSON—an enormous advantage. On the negative side, the Neanes data format carries some legacy cruft and reflects internal implementation details, making it unsuitable as a general-purpose data format for other Byzantine notation scorewriters.

Here there is a lesson to be learned from the Finale to Dorico transition regarding the importance of open, vendor-neutral data formats—just as MusicXML created a path to export data from Finale and import data into Dorico, so too could a vendor-neutral data format for Byzantine notation scorewriters support e.g. export from Melodos and e.g. import into Neanes.

**Comment by basil** [comment]
This is just software, so anything is possible given sufficient time and effort. As with any open-source software project, you are welcome to create a pull request that implements whatever you desire.

**Comment by cnkaram** [comment]
Thank you for your response. I did submit that request a while ago.

Chadi 
 

    On Saturday, September 14, 2024 at 11:17:19 AM CDT, Basil Crow ***@***.***> wrote:  
 
 


This is just software, so anything is possible given sufficient time and effort. As with any open-source software project, you are welcome to create a pull request that implements whatever you desire.

**Comment by basil** [comment]
You have not submitted a pull request to this GitHub organization, but we would welcome contributions from you or anyone else to improve the software.

**Comment by danielgarthur** [comment]
A while ago I did make several attempts to investigate the feasibility of exporting to Word, but did not see any clear way of doing it. Creating a Word document is simple enough, but getting Word to render precisely what was designed in Neanes is difficult because I have to fight against Word's justification algorithms, among other things.

But perhaps someone else will come along and figure out a clever way of doing it. This is what Basil is referring to. Even though I wrote most of the software, the source code is freely available and anyone can modify it and submit a request for their changes to become part of the program that is released. If someone wrote code to support export to Word, I would certainly approve it and it would become part of Neanes. But I don't think I will spend more time on it because it doesn't seem like a good way forward for the reasons Basil already explained.

**Comment by cnkaram** [comment]
Ok. Thank you for getting back to me on this. I understand the difficulties that you just described with Word.


Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, Sep 14, 2024 at 12:49 PM, ***@***.***> wrote:   


A while ago I did make several attempts to investigate the feasibility of exporting to Word, but did not see any clear way of doing it. Creating a Word document is simple enough, but getting Word to render precisely what was designed in Neanes is difficult because I have to fight against Word's justification algorithms, among other things.

But perhaps someone else will come along and figure out a clever way of doing it. This is what Basil is referring to. Even though I wrote most of the software, the source code is freely available and anyone can modify it and submit a request for their changes to become part of the program that is released. If someone wrote code to support export to Word, I would certainly approve it and it would become part of Neanes. But I don't think I will spend more time on it because it doesn't seem like a good way forward for the reasons Basil already explained.
